test/unit/org/apache/cassandra/db/guardrails/GuardrailPageSizeTest.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.cassandra.cql3.QueryProcessor;
+import org.apache.cassandra.db.ConsistencyLevel;
+import org.apache.cassandra.service.ClientState;
+import org.apache.cassandra.service.QueryState;
+import org.apache.cassandra.transport.ProtocolVersion;
+
+import static java.lang.String.format;
+
+/**
+ * Tests the guardrail for the page size, {@link Guardrails#pageSize}.
+ */
+public class GuardrailPageSizeTest extends ThresholdTester
+{
+    private static final int PAGE_SIZE_WARN_THRESHOLD = 5;
+    private static final int PAGE_SIZE_ABORT_THRESHOLD = 10;
+
+    public GuardrailPageSizeTest()
+    {
+        super(PAGE_SIZE_WARN_THRESHOLD,
+              PAGE_SIZE_ABORT_THRESHOLD,
+              DatabaseDescriptor.getGuardrailsConfig().getPageSize(),
+              Guardrails::setPageSizeThreshold,
+              Guardrails::getPageSizeWarnThreshold,
+              Guardrails::getPageSizeAbortThreshold);
+    }
+
+    @Before
+    public void setupTest()
+    {
+        createTable("CREATE TABLE IF NOT EXISTS %s (k INT, c INT, v TEXT, PRIMARY KEY(k, c))");
+    }
+
+    @Test
+    public void testSelectStatementAgainstPageSize() throws Throwable
+    {
+        assertPagingValid("SELECT * FROM %s", 3);
+        assertPagingValid("SELECT * FROM %s", PAGE_SIZE_WARN_THRESHOLD);
+
+        assertPagingWarns("SELECT * FROM %s", 6, format("Query for table %s with page size %s exceeds warning threshold of %s.", currentTable(), 6, PAGE_SIZE_WARN_THRESHOLD));
+        assertPagingWarns("SELECT * FROM %s", PAGE_SIZE_ABORT_THRESHOLD, format("Query for table %s with page size %s exceeds warning threshold of %s.", currentTable(), PAGE_SIZE_ABORT_THRESHOLD, PAGE_SIZE_WARN_THRESHOLD));
+
+        assertPagingAborts("SELECT * FROM %s", 11, format("Aborting %s table query, page size %s exceeds abort threshold of %s.", currentTable(), 11, PAGE_SIZE_ABORT_THRESHOLD));
+    }
+
+    @Test
+    public void testIgnoreInternalClients() throws Throwable
+    {
+        assertPagingIgnored("SELECT * FROM %s", PAGE_SIZE_WARN_THRESHOLD);
+        assertPagingIgnored("SELECT * FROM %s", PAGE_SIZE_ABORT_THRESHOLD);
+    }
+
+    private void assertPagingValid(String query, int pageSize) throws Throwable
+    {
+        assertValid(() -> executeWithPaging(userClientState, query, pageSize));
+    }
+
+    private void assertPagingIgnored(String query, int pageSize) throws Throwable
+    {
+        assertValid(() -> executeWithPaging(superClientState, query, pageSize));
+        assertValid(() -> executeWithPaging(systemClientState, query, pageSize));
+    }
+
+    private void assertPagingWarns(String query, int pageSize, String message) throws Throwable
+    {
+        assertWarns(() -> executeWithPaging(userClientState, query, pageSize), message);
+    }
+
+    private void assertPagingAborts(String query, int pageSize, String message) throws Throwable
+    {
+        assertAborts(() -> executeWithPaging(userClientState, query, pageSize), message);
```

Comment:
Same as before, we can get rid of the `message` argument:
```suggestion
        assertAborts(() -> executeWithPaging(userClientState, query, pageSize),
                     format("Aborting query for table %s, page size %s exceeds abort threshold of %s.",
                            currentTable(), pageSize, PAGE_SIZE_ABORT_THRESHOLD));
```

Reply:
fixed, it looks like this right now 
```java
    private void assertPagingWarns(String query, int pageSize) throws Throwable
    {
        assertWarns(() -> executeWithPaging(userClientState, query, pageSize),
                    format("Query for table %s with page size %s exceeds warning threshold of %s.", currentTable(), pageSize, PAGE_SIZE_WARN_THRESHOLD));
    }

    private void assertPagingAborts(String query, int pageSize) throws Throwable
    {
        assertAborts(() -> executeWithPaging(userClientState, query, pageSize),
                     format("Aborting query for table %s, page size %s exceeds abort threshold of %s.", currentTable(), pageSize, PAGE_SIZE_ABORT_THRESHOLD));
    }
```